internal/app/compress/gzip-writer.go

```diff
@@ -0,0 +1,66 @@
+package compress
+
+import (
+	"compress/gzip"
+	"io"
+	"net/http"
+)
+
+type GzipWriter struct {
+	w  http.ResponseWriter
+	zw *gzip.Writer
+}
+
+func NewGzipWriter(w http.ResponseWriter) *GzipWriter {
+	return &GzipWriter{
+		w:  w,
+		zw: gzip.NewWriter(w),
+	}
+}
+
+func (c *GzipWriter) Header() http.Header {
+	return c.w.Header()
+}
+
+func (c *GzipWriter) Write(p []byte) (int, error) {
+	return c.zw.Write(p)
+}
+
+func (c *GzipWriter) WriteHeader(statusCode int) {
+	if statusCode < 300 {
```

Comment:
```suggestion
	if statusCode < 300 {
```
This is a magic number.